src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -184,6 +188,7 @@ private void switchTimelines(final Timeline activeTimeline, final Timeline nextT
                     final Timeline.StoragePosition storagePosition =
                             topicRepositoryHolder.createStoragePosition(activeTimeline);
                     activeTimeline.setLatestPosition(storagePosition);
+                    scheduleTimelineCleanup(activeTimeline);
```

Comment:
Why?
When we are at this point the topic is already created.
So I thought that it will be a good place to add this when we update the previous timeline.